[PL-49148] Add option to Configure an optional registry mirror for delegate images (#6384)
harness/developer-hub: docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md

   The schedule change for CronJob will take effect immediately, and the next upgrade run will follow the new schedule. If you have made any other changes to the YAML file, such as updating the image, configuration, environment variables, and so on, those changes will take effect during the next run.

### Configure an optional registry mirror for delegate images

If you use Docker pull through registry cache (`https://docs.docker.com/docker-hub/mirror/`), you can configure `upgrader` to use an optional registry mirror for your delegate images.

> When this feature is configured, Harness Delegate images won't be fetched from the public Docker Hub but from the designated mirror.

Reviewer: ```suggestion
When this feature is configured, Harness Delegate images are fetched from the designated mirror, instead of public Docker Hub.
```

Alternate suggestion:

```
Configure this feature to fetch Harness Delegate images from a designated mirror instead of public Docker Hub.
```

```yaml
mode: Delegate
dryRun: false
workloadName: delegate-name
namespace: harness-delegate-ng
containerName: delegate
registryMirror: us.gsr.io/gcr-mirror
delegateConfig:
  accountId: <YOUR_ACCOUNT_ID>
  managerHost: <MANAGER_HOST>
```

> During an upgrade, when upgrader seeks to update the delegate to `harness/delegate:verno`, it will utilize the image from `us.gsr.io/gcr-mirror/harness/delegate:verno`.

Author: ```suggestion
During an upgrade, when `upgrader` seeks to update the delegate to `harness/delegate:verno`, it will utilize the image from `us.gsr.io/gcr-mirror/harness/delegate:verno`.
```
